src/m2n/M2N.cpp

```diff
@@ -222,7 +223,12 @@ void M2N::send(
       _masterCom->receive(ack, 0);
       _masterCom->send(ack, 0);
     }
-    Event e("m2n.sendData", precice::syncMode);
+
+    if (withGradient)
+      Event e("m2n.sendGradientData", precice::syncMode);
+    else
+      Event e("m2n.sendData", precice::syncMode);
```

Comment:
You won't be able to measure here anything. As stated in the [Eventtimings](https://github.com/precice/EventTimings#readme), the framework measures the time until destruction. In this case, the object is destroyed immediately after leaving the if branch. I guess you can simply use a ternary operator as a remedy.

Reply:
In case the event doesn't allow for it, you could also parametrize the event name ("send(Gradient)Data") and create it later using the parametrized name.